src/herder/HerderSCPDriver.cpp

```diff
@@ -722,4 +728,97 @@ HerderSCPDriver::acceptedCommit(uint64_t slotIndex, SCPBallot const& ballot)
 {
     mSCPMetrics.mAcceptedCommit.Mark();
 }
+
+void
+HerderSCPDriver::setMetricStartTime(uint64_t slotIndex,
+                                    VirtualClock::time_point start,
+                                    bool isNomination)
+{
+    SCPTimingPtr timing;
```

Comment:
don't use `shared_ptr` here, then you can simplify the code to just be:

`auto &timing = mSCPExecutionTimes[slotIndex];`